File: OutOfOffice.Contracts/Models/Project.cs
namespace OutOfOffice.Contracts.Models;

public class Project
{
    public int Id { get; set; }
    public int ProjectTypeId { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public int ProjectManagerId { get; set; }
    public int StatusId { get; set; }
    public string Comment { get; set; }

    public Employee ProjectManager { get; set; }
    public ProjectType ProjectType { get; set; }
    public ProjectStatus Status { get; set; }
}

File: outofoffice.client/src/app/models/project.ts
import { Employee } from './employee'

export interface Project {
  id: number
  startDate: string
  endDate: string | null
  comment: string
  projectManager: Employee
  projectType: ProjectType
  status: ProjectStatus
  projectEmployees: ProjectEmployee[]
}

export interface ProjectType {
  id: number
  name: string
}

export interface ProjectStatus {
  id: number
  name: string
}

export interface ProjectPayload {
  startDate: string
  endDate: string | null
  comment: string
  projectManagerId: number
  projectTypeId: number
  statusId: number
}

export interface ProjectEmployee {
  startDate: string
  endDate: string | null
  employeeId: number
  projectId: number
}
